### ISSUE TITLE
[Bug]: File caption is shown instead of file title

### ISSUE THREAD

**Issue by prototyperspective** [commons-app/apps-android-commons#6023]
### Summary

In [this category](https://commons.wikimedia.org/wiki/Category:Spoken_Wikipedia_articles_using_English-language_speech_synthesis) of audio files the file captions are shown instead of the file title (it's the example I also used for #5922 as the cat I'd bookmark and then use the app mostly for once that is possible)

### Steps to reproduce

1. Go the a category which can be done by searching for a file and then tapping on one of it's cats since the patch that fixes category search is still not released – in this case I searched for "Spoken Wikipedia" to get to this cat of audio files
2. Scroll to see it's all just audio captions if a caption is set

### Expected behaviour

It should show the file title instead of the caption or the file title and some bits of the caption.

### Actual behaviour

(Only) the file caption is shown. File titles on Commons should be descriptive, see [Commons:File naming](https://commons.wikimedia.org/wiki/Commons:File_naming).

### Android version

Android 11

### Commons app version

5.0.2

### Would you like to work on the issue?

Prefer not

**Comment by nicolas-raoul** [comment]
Thanks for bringing this up!

This is intended, as captions must be concise and descriptive of the file's content.
File names often contain a lot of other information, such as author name and numbers, which makes them less readable by humans.

https://commons.wikimedia.org/wiki/Commons:File_captions says:
> They are meant to store short, multilingual descriptions about files.
> [...]
> For example, [File:Fire station Hallstatt - October 2017 - 02.jpg](https://commons.wikimedia.org/wiki/File:Fire_station_Hallstatt_-_October_2017_-_02.jpg) has the English caption "Hallstatt fire station in October 2017".

Captions are thus much more user friendly than file titles, especially on small screens such as those on mobile devices.

Additionally, many file titles are in Japanese or Arabic etc, with captions in various languages making them readable by people not familiar with these languages.

It seems that the category you mentioned contains files uploaded by you.
You will thus be in the best position to fix the captions following the documentation above.
For instance for https://commons.wikimedia.org/wiki/File:Wikipedia_-_Nature_Restoration_Law_(spoken_by_AI_voice).mp3 I see that the caption is currently `Article version 30 Oct. 2024`, which is obviously not great. A much better caption would be for instance: `Wikipedia article "Nature Restoration Law" spoken by AI voice`. I know it is a lot of work, but it is the best thing to do. :-)

By the way thanks a lot for uploading these important articles, and for clearly mentioning that they are spoken by AI.
Cheers!

**Comment by prototyperspective** [comment]
Please reopen! It's not feasible to develop a separate app for Commons and this is critically needed. File titles are also shown for images and there is no good reason to not show them for audio files. You could add an option to only see captions but not file titles. As the link explains, file-titles are meant to be usually descriptive. Do you not have much experience on Commons or how do you not know that many files have very low-quality captions? [Example](https://commons.wikimedia.org/wiki/File:World_map_of_disease_burden_rates_from_cancers,_2019,_OWID.png) [example](https://commons.wikimedia.org/wiki/File:Total-oda-for-water-supply-and-sanitation-by-recipient.png) [example](https://commons.wikimedia.org/wiki/File:Prevalence-of-anorexia-vs-gdp-per-capita.png) [example](https://commons.wikimedia.org/wiki/File:Lung_cancer_deaths_per_100_000,_United_States,_1950_to_2020,_OWID.png). What you said is not true.

**Comment by nicolas-raoul** [comment]
> there is no good reason to not show them for audio files.

You are right that we should strive towards coherence between all types of media.

I believe we are currently coherent:

# Picture

![Screenshot_20241212-100258~3.png](https://github.com/user-attachments/assets/83138889-7e1c-405b-9a32-fe0395799b7a)

![Screenshot_20241212-100307~2.png](https://github.com/user-attachments/assets/68e0039f-cb04-45a5-ab32-3eebd5c95eb9)

# Audio

![Screenshot_20241212-100336.png](https://github.com/user-attachments/assets/9ecd1fed-44d9-4b45-b00f-979d18282d43)

![Screenshot_20241212-095759~2.png](https://github.com/user-attachments/assets/143b9f07-9bed-4109-b0b6-4fdd6bf37675)

**Comment by prototyperspective** [comment]
Thanks for your response. First of all it may helpful to other readers of this issue to make those screenshots smaller as described [here](https://gist.github.com/stevecondylios/dcadb4fc73e63f27a3bbcf17e52058bf). It's very hard to read threads with huge images in between.

You did not address any of the other points I made. That file titles are also shown for images was just one of them. If they aren't I haven't used the app long enough (because I would use it mostly to watch videos like the YouTube app or listen to those audios above which is not possible) to notice the many large problems this causes. Or are they only not shown if the filetitle is detected to be of some other language? (That would make sense.) In the case that they aren't shown if a caption is set on the image, please also show the file title for images in addition. I argued for showing the file title as the title and the caption has its subheader but one could also reverse it to show the caption as main title if one is set and the file-title in a smaller font beneath. That would be useful in any case. (I still think the succinct file title should be shown first as it's maxlength is shorter.)

Please reopen so other devs & users can see this and participate. I think many would find seeing the title useful or needed and there's many ways this could be implemented, like e.g. showing only the first x chars of the caption by default and the full caption when the caption is tapped.

I've come across countless files with either bad captions or captions that are only useful when in addition to the file title (like subheaders) as it is the case in Wikipedia (MediaViewer). I only named a few examples but there is an unfixable large number of them and that's fine – just show both and it's not a problem. For example open this category in the app [Sanitation statistics](https://commons.wikimedia.org/wiki/Category:Sanitation_statistics).

In short: consistency between file-types was not my only point and as this problem exists for images as well please also show the file-titles for images somehow (there could be an option in the settings to just have the captions displayed).

**Comment by prototyperspective** [comment]
I think many users understand caption to be something like subheaders or generally complementary to the file title. Regardless whether or not many understand it that way, lots of files have captions like that that are complementary to the file title and/or not descriptive without the title or secondary to it (e.g. people wonder why they would put information already in the file title also into the caption so add some other info there). [Another example](https://commons.wikimedia.org/wiki/Category:2023_in_science) but there's too many cats & files to list them all and that obviously also goes changing all of those files' captions which will never be done and is infeasible and also not needed. There could be an option to make it only display the caption or it could display the caption first and the file title beneath it rather than the other way around, but please enable seeing file-titles – I guess not many people are using the app much for browsing  but if watching videos & listening to audios become possible more may use it which is when it should be useful in the real world (i.e. not based on some hypothetical ideal how things _should_ be but about how things actually _are_ and that ideal is not widely shared it seems to begin with either).

**Comment by prototyperspective** [comment]
There's too many cases of files and category with lots of files where the caption wasn't thought of and written as the singular file title. Very often, it's like a subheading or some text just adding further info to the file title (like date and place of an event) that should be very visible to somebody going to the file page. This isn't some kind of rare exception, it's very common across Commons and not showing the file title does nothing short of breaking the app.

I think most contributors never thought of it as being the file title, also because that would be redundant as most enter a descriptive file title, [as is now required per Commons policy](https://commons.wikimedia.org/wiki/Commons:File_naming).

It's really very important that the file title is also displayed which of the two is displayed first/larger doesn't play a big role and there can also be a setting to disable either from showing. It's really of critical importance to the app. Please reopen, I'll edit the title to make it not say it's a bug.
